libbindgen/tests/tests.rs

```
@@ -0,0 +1,132 @@
+extern crate clap;
+extern crate diff;
+#[macro_use]
+extern crate env_logger;
+extern crate libbindgen;
+extern crate log;
+extern crate shlex;
+
+use std::env;
+use std::fs;
+use std::io::{BufRead, BufReader, Error, ErrorKind, Read};
+use std::path::{Path, PathBuf};
+
+#[path="../../src/options.rs"]
+mod options;
+use options::builder_from_flags;
+
+fn compare_generated_header(header: &PathBuf,
+                            builder: libbindgen::Builder)
+                            -> Result<(), Error> {
+    let file_name = try!(header.file_name()
+        .ok_or(Error::new(ErrorKind::Other, "spawn_bindgen expects a file")));
+
+    let mut expected = PathBuf::from(header);
+    expected.pop();
+    expected.pop();
+    expected.push("expectations");
+    expected.push("tests");
+    expected.push(file_name);
+    expected.set_extension("rs");
+
+    let output = match builder.generate() {
+        Ok(bindings) => bindings.to_string(),
+        Err(_) => "".to_string(),
```

Probably we shouldn't ignore this error?